src/sentry/rules/processor.py

```diff
@@ -170,17 +170,38 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
         rule_condition_list = rule.data.get("conditions", ())
         frequency = rule.data.get("frequency") or Rule.DEFAULT_FREQUENCY
 
+        self.logger.info("apply_rule", extra={"rule_id": rule.id})
         try:
             environment = self.event.get_environment()
+            self.logger.info(
+                "apply_rule got environment", extra={"environment": environment, "rule_id": rule.id}
+            )
         except Environment.DoesNotExist:
+            self.logger.info("apply_rule environment does not exist", extra={"rule_id": rule.id})
             return
 
         if rule.environment_id is not None and environment.id != rule.environment_id:
+            self.logger.info(
+                "apply_rule environment does not match",
+                extra={
+                    "rule_id": rule.id,
+                    "rule_environment_id": rule.environment_id,
+                    "event_environment_id": environment.id,
+                },
+            )
             return
 
         now = timezone.now()
         freq_offset = now - timedelta(minutes=frequency)
         if status.last_active and status.last_active > freq_offset:
+            self.logger.info(
+                "apply_rule skipping rule because of last_active",
+                extra={
+                    "rule_id": rule.id,
+                    "last_active": status.last_active,
+                    "freq_offset": freq_offset,
+                },
+            )
```

Comment:
Probably worth logging `self.group.id` and maybe `self.event.id` as well.

Actually, would be really helpful to also log `self.is_(new/regression/new_group_environment` and `has_reappeared`